File: src/modifiers/headerProminence.ts
import { createModifier } from './createModifier';

/// SwiftUI `Prominence` values for section headers.
export type Prominence = 'standard' | 'increased';

/// SwiftUI `.headerProminence(_:)`. Sets the prominence of section headers
/// within the view — `'increased'` gives list section headers a larger,
/// more emphasized appearance.
export function headerProminence(value: Prominence) {
  return createModifier('headerProminence', { value });
}

File: apple/Sources/ReactNativeWatchOS/Modifiers/HeaderProminence.swift
import SwiftUI

/// SwiftUI `.headerProminence(_:)`. Sets the `Prominence` of section headers
/// in the view — `.increased` emphasizes list section headers.
enum RNWHeaderProminenceModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("headerProminence") { view, params, _ in
            AnyView(view.headerProminence(parse(params.string("value"))))
        }
    }

    private static func parse(_ s: String?) -> Prominence {
        s == "increased" ? .increased : .standard
    }
}
